resources/build/bootstrap-ubuntu-20.04.sh

```diff
@@ -5,7 +5,7 @@
 set -xeo pipefail
 sudo apt-get update
 # Install gcc, clang-tidy and Python 3.
-sudo apt-get install -y build-essential clang-tidy-12 python3-pip
+sudo apt-get install -y build-essential pkg-config clang-format-12 clang-tidy-12 python3-pip
```

Comment:
[consider][minor] Might be worth calling out why `pkg-config` is required. I would say put it with the big list below, but that doesn't quite match since "These would be installed as part of the conan package" clearly isn't true for `pkg-config`.

Reply:
Wording updated [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/429/commits/d2a882f910b536ed9c480cdd8cdf7dfffc0dd930).